src/Common/MessageBroker/Integration/AmqpLib/AmqpPublisher.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\MessageBroker\Integration\AmqpLib;
+
+use Gaming\Common\MessageBroker\Exception\MessageBrokerException;
+use Gaming\Common\MessageBroker\Model\Message\Message;
+use Gaming\Common\MessageBroker\Publisher;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Message\AMQPMessage;
+use Throwable;
+
+final class AmqpPublisher implements Publisher
+{
+    private ?AMQPChannel $channel;
+
+    private AMQPMessage $message;
+
+    public function __construct(
+        private readonly ConnectionFactory $connectionFactory,
+        private readonly Topology $topology,
+        private readonly ReliablePublishing $reliablePublishing,
+        private readonly string $exchangeToPublishTo,
+        int $deliveryMode
+    ) {
+        $this->channel = null;
+        $this->message = new AMQPMessage(
+            '',
+            [
+                'delivery_mode' => $deliveryMode
+            ]
+        );
+    }
+
+    public function publish(Message $message): void
+    {
+        $this->channel ??= $this->createConfirmingChannelAndTopology();
+
+        $this->message->setBody(
+            json_encode(
+                [(string)$message->name(), $message->body()],
+                JSON_THROW_ON_ERROR
+            )
+        );
+
+        try {
+            $this->channel->basic_publish(
```

Comment:
This (+ `new AMQPMessage()` and `setBody()`) should be abstracted in a similar way as `ReliablePublishing`. This way `mandatory`, other flags as well as message properties can be configurable without providing a completely new implementation of the `Publisher` interface and without blowing this class up.